### PR TITLE
For issue #15934 fix ConcurrentModificationException on WifiConnectionMonitor

### PR DESCRIPTION
**Pull request by Amejia481**
It looks like we were [iterating through](https://docs.oracle.com/javase/7/docs/api/java/util/ConcurrentModificationException.html) the `callbacks` list while it was modified by another thread, creating defensive copy while looping should fix the issue, I tested but I don't have the exact STR, but while testing I wasn't able to reproduce. Creating test for this edge case it's a bit tricky, for this reason I didn't include it.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/wifi/WifiConnectionMonitor.kt [hunk 1]
```diff
@@ -35,16 +35,21 @@ class WifiConnectionMonitor(app: Application) {
...
 
         override fun onAvailable(network: Network?) {
-            callbacks.forEach { it(true) }
+            notifyListeners(true)
             lastKnownStateWasAvailable = true
         }
     }
 
+    internal fun notifyListeners(value: Boolean) {
+        val items = ArrayList(callbacks)
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/wifi/WifiConnectionMonitor.kt:49]: Can we instantiate this (line 29) as an ArrayList? Or are you doing this here to make it non-mutable?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/wifi/WifiConnectionMonitor.kt:49]: [MutableList](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-mutable-list/) is a more abstract interface, and could be swapped by other implementations, under the hood is backed by a `ArrayList` when we call `mutableListOf` :)

image

Here we want to create a defensive copy of the initial list, to iterate safely while the original list can be modified.

Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/wifi/WifiConnectionMonitor.kt:49]: Excellent! Thank you :D

**Review by eliserichards — APPROVED** [review]
Once the CI passes this is good to go! 🚀
